eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelData.scala

```diff
@@ -215,7 +215,7 @@ object HtlcResult {
   case class RemoteFailMalformed(fail: UpdateFailMalformedHtlc) extends Fail
   case class OnChainFail(cause: Throwable) extends Fail
   case object ChannelFailureBeforeSigned extends Fail
-  case class DisconnectedBeforeSigned(channelUpdate: ChannelUpdate) extends Fail { assert(!Announcements.isEnabled(channelUpdate.channelFlags), "channel update must have disabled flag set") }
+  case class DisconnectedBeforeSigned(channelUpdate: ChannelUpdate) extends Fail { assert(channelUpdate.channelFlags.disable, "channel update must have disabled flag set") }
```

Comment:
nit: why do we use `assert` instead of `require` here?

Reply:
Have no idea, will fix

Reply:
Done in 130f44772.